pkg/injector/webhook_test.go

```diff
@@ -93,3 +97,288 @@ func (mc mockCertificate) GetCertificateChain() []byte           { return []byte
 func (mc mockCertificate) GetPrivateKey() []byte                 { return []byte("key") }
 func (mc mockCertificate) GetIssuingCA() []byte                  { return []byte("ca") }
 func (mc mockCertificate) GetExpiration() time.Time              { return time.Now() }
+
+var _ = Describe("Testing isAnnotatedForInjection", func() {
+	Context("when the inject annotation is one of enabled/yes/true", func() {
+		It("should return true to enable sidecar injection", func() {
+			annotation := map[string]string{constants.SidecarInjectionAnnotation: "enabled"}
+			exists, enabled, err := isAnnotatedForInjection(annotation)
+			Expect(exists).To(BeTrue())
+			Expect(enabled).To(BeTrue())
+			Expect(err).To(BeNil())
+		})
+
+		It("should return true to enable sidecar injection", func() {
+			annotation := map[string]string{constants.SidecarInjectionAnnotation: "yes"}
+			exists, enabled, err := isAnnotatedForInjection(annotation)
+			Expect(exists).To(BeTrue())
+			Expect(enabled).To(BeTrue())
+			Expect(err).To(BeNil())
+		})
+
+		It("should return true to enable sidecar injection", func() {
+			annotation := map[string]string{constants.SidecarInjectionAnnotation: "true"}
+			exists, enabled, err := isAnnotatedForInjection(annotation)
+			Expect(exists).To(BeTrue())
+			Expect(enabled).To(BeTrue())
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Context("when the inject annotation is one of disabled/no/false", func() {
+		It("should return false to disable sidecar injection", func() {
+			annotation := map[string]string{constants.SidecarInjectionAnnotation: "disabled"}
+			exists, enabled, err := isAnnotatedForInjection(annotation)
+			Expect(exists).To(BeTrue())
+			Expect(enabled).To(BeFalse())
+			Expect(err).To(BeNil())
+		})
+
+		It("should return false to disable sidecar injection", func() {
+			annotation := map[string]string{constants.SidecarInjectionAnnotation: "no"}
+			exists, enabled, err := isAnnotatedForInjection(annotation)
+			Expect(exists).To(BeTrue())
+			Expect(enabled).To(BeFalse())
+			Expect(err).To(BeNil())
+		})
+
+		It("should return false to disable sidecar injection", func() {
+			annotation := map[string]string{constants.SidecarInjectionAnnotation: "false"}
+			exists, enabled, err := isAnnotatedForInjection(annotation)
+			Expect(exists).To(BeTrue())
+			Expect(enabled).To(BeFalse())
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Context("when the inject annotation does not exist", func() {
+		It("should return false to indicate the annotation does not exist", func() {
+			annotation := map[string]string{}
+			exists, _, _ := isAnnotatedForInjection(annotation)
+			Expect(exists).To(BeFalse())
```

Comment:
Could we go ahead and also test the second and third return vars (enabled and err) are set to false and nil?

Reply:
Done.